File: raccoon-cloud/raccoon-cloud-system/src/main/java/com/raccoon/cloud/system/cmms/entity/InspectionWorkOrder.java
package com.raccoon.cloud.system.cmms.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 巡检工单主表 inspection_work_order。
 */
@Data
@TableName("inspection_work_order")
public class InspectionWorkOrder {
    @TableId(type = IdType.AUTO)
    private Long id;
    /** 工单编号 */
    private String orderNo;
    /** 巡检区域 */
    private String area;
    /** 班次 1早班 2中班 3夜班 */
    private Integer shiftType;
    private Long inspectorId;
    private String inspectorName;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime planStartTime;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime planEndTime;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime actualStartTime;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime actualEndTime;
    /**
     * 1待下发 2待执行 3执行中 4已完成 5已取消
     */
    private Integer status;
    private Long createBy;
    private String createByName;
    private Long updateBy;
    /** 来源巡检计划 */
    private Long planId;
    /** 关联巡检任务 */
    private Long taskId;
    private String remark;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    private LocalDateTime createTime;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    private LocalDateTime updateTime;
}

File: raccoon-ui/src/api/inspectionWorkOrder.ts
import request from '@/utils/request'

export interface IwoStepDraft {
  stepOrder: number
  type: string
  target?: string
  description?: string
  deviceId?: number | null
  deviceName?: string
  checkItem?: string
  standardMin?: number
  standardMax?: number
  unit?: string
  detectionMethod?: string
}

export interface IwoCreateBody {
  area: string
  shiftType: number
  inspectorId: number
  inspectorName?: string
  planStartTime: string
  planEndTime: string
  remark?: string
  planId?: number
  taskId?: number
  steps: IwoStepDraft[]
}

export interface IwoRow {
  id: number
  orderNo: string
  area: string
  shiftType: number
  inspectorId: number
  inspectorName?: string
  planStartTime: string
  planEndTime: string
  actualStartTime?: string
  actualEndTime?: string
  status: number
  planId?: number
  taskId?: number
  remark?: string
}

export interface IwoFull {
  id: number
  orderNo: string
  area: string
  shiftType: number
  inspectorId: number
  inspectorName?: string
  planStartTime: string
  planEndTime: string
  actualStartTime?: string
  actualEndTime?: string
  status: number
  createBy?: number
  createByName?: string
  planId?: number
  taskId?: number
  remark?: string
  steps: IwoStepVO[]
}

export interface IwoStepVO {
  id: number
  orderId: number
  stepOrder: number
  type: string
  target?: string
  description?: string
  deviceId?: number
  deviceName?: string
  deviceCode?: string
  voltageLevel?: string
  checkItem?: string
  standardMin?: number
  standardMax?: number
  unit?: string
  detectionMethod?: string
  actualValue?: string
  isException?: number
  collectTime?: string
  photoUrl?: string
  remark?: string
}

export const iwoCreate = (data: IwoCreateBody) =>
  request({ url: '/cmms/inspectionWorkOrder/create', method: 'post', data })

export const iwoIssue = (id: number) =>
  request({ url: '/cmms/inspectionWorkOrder/issue', method: 'post', data: { id } })

export const iwoCancel = (id: number) =>
  request({ url: '/cmms/inspectionWorkOrder/cancel', method: 'post', data: { id } })

export const iwoPage = (data: {
  page?: number
  size?: number
  area?: string
  shiftType?: number
  status?: number
  planId?: number
  taskId?: number
  planStartFrom?: string
  planStartTo?: string
}) => request({ url: '/cmms/inspectionWorkOrder/page', method: 'post', data })

export const iwoMyPage = (data: { page?: number; size?: number; tab: string; inspectorId?: number }) =>
  request({ url: '/cmms/inspectionWorkOrder/myPage', method: 'post', data })

export const iwoDetail = (orderId: number) =>
  request({ url: '/cmms/inspectionWorkOrder/detail', method: 'get', params: { orderId } })

export const iwoStepExecute = (data: { detailId: number; remark?: string }) =>
  request({ url: '/cmms/inspectionWorkOrder/step/execute', method: 'post', data })

export const iwoStepCollect = (data: { detailId: number; actualValue: string; photoUrl?: string; remark?: string }) =>
  request({ url: '/cmms/inspectionWorkOrder/step/collect', method: 'post', data })

export const iwoScanDevice = (data: { detailId: number; deviceCode: string }) =>
  request({ url: '/cmms/inspectionWorkOrder/scanDevice', method: 'post', data })
